libazureinit/src/config.rs

```diff
@@ -421,6 +482,40 @@ impl Config {
     }
 }
 
+/// Checks the first line of `file_path` for the `#azure-init-config` header.
+/// Returns an error if the file doesn't start with `#azure-init-config`.
+fn check_azure_init_header(file_path: &Path) -> Result<(), Error> {
+    if !file_path.is_file() {
+        // If it's not a file, ignore or skip the check
+        return Ok(());
+    }
+
+    let file = fs::File::open(file_path).map_err(|e| {
+        tracing::error!("Failed to open file {:?}: {:?}", file_path, e);
+        Error::Io(e)
+    })?;
+    let mut reader = std::io::BufReader::new(file);
+
+    let mut first_line = String::new();
+    let _ = reader.read_line(&mut first_line).map_err(|e| {
+        tracing::error!("Error reading first line of {:?}: {:?}", file_path, e);
+        Error::Io(e)
+    })?;
+
+    if !first_line.trim_start().starts_with("#azure-init-config") {
```

Comment:
i would suggest allowing any whitespace in between `#` and `aure-init-config` because I bet many will try `# azure-init-config`

Reply:
trim the end too.